lokimq/bt_serialize.h

```diff
@@ -909,5 +922,192 @@ class bt_dict_consumer : private bt_list_consumer {
     bt_dict_consumer consume_dict_consumer() { return consume_dict_data(); }
 };
 
+/// Currently this is not idiot-proof.  That is to say, the compiler won't catch everything and
+/// you technically can create invalid BEncoded strings.  Basically you can add list elements
+/// to a dict and dict elements to a list, and obviously that can end badly.  This may be fixed
+/// at a later time.
+struct fixed_buffer_producer
+{
+    char* begin;
+    char* end;
+
+    char* original_begin;
```

Comment:
The dict version also needs to have a `std::string_view last_key` pointing at the already-written previous key value so that it can throw if you try to add out of order.